src/lambda_function/raw_sync/README.md

```diff
@@ -0,0 +1,38 @@
+# Raw Sync Lambda
+
+The raw sync Lambda ensures that the input and raw S3 buckets are synchronized
+by verifying that all non-zero sized JSON in each export in the input bucket
+have a corresponding object in the raw bucket.
+
+If a JSON file from an export is found to not have a corresponding object in the raw bucket,
+the file is submitted to the raw Lambda (via the dispatch SNS topic) for processing.
```

Comment:
That's a possibility. Assuming the raw Lambda is working correctly, that could mean a few things:

1. The raw Lambda was throttled, or it failed for normal resource constraint reasons, in which case we do want to re-submit. 
2. The file is unable to be processed by the raw Lambda in its current configuration. Probably this means we've encountered an extremely large JSON file, probably ~120-150 GB uncompressed, or 3 times the size of anything we've received before. If we don't handle this case, the consequences are that we will submit this behemoth for processing once a day where it will timeout after 15 minutes. Alternatively, perhaps a non-zip file or some other corrupted file is accidentally deposited in the input bucket, which would be ignored by the raw-sync Lambda after it determines it's not a zip-file (although this process would also repeat each day until the corrupted file is removed). We don't want to re-submit the file, but I can't think of a graceful way for the raw-sync lambda to differentiate between this case and case (1).

In short... I can't think of a way for the raw-sync Lambda to gracefully handle files which simply can't be processed by raw Lambda. It could help to have raw-sync Lambda publish to an SNS topic which sends us an email when it encounters a file which is out-of-sync, at which point we could manually intervene to see what happened. In the interim we could watch the raw Lambda's DLQ to see if there are any files which repeatedly fail to be processed.

Reply:
I can see how the 2nd scenario would be difficult to catch. I don't remember the retry limit but yeah if it ends up in the DLQ, that can be a good area to check in the interim.